firefox.launcher

```diff
@@ -52,4 +52,7 @@ if [ ! -d "$DOTMOZILLA" ]; then
   fi
 fi
 
+# The desktop file is exported by snapd as firefox_firefox.desktop
+export MOZ_APP_REMOTINGNAME=firefox_firefox
```

Comment:
Cf https://github.com/canonical/firefox-snap/commit/b5a8d62e9d5e3585773c957980ad34d856787cba and thus https://bugzilla.mozilla.org/show_bug.cgi?id=1897907

Reply:
> it should probably be changed within the snapcraft.yaml itself rather than here

You're absolutely right, fixed 😅

> I'm worried changing this var might have unexpected side effects, probably more 
> Cf https://github.com/canonical/firefox-snap/commit/b5a8d62e9d5e3585773c957980ad34d856787cba and thus https://bugzilla.mozilla.org/show_bug.cgi?id=1897907

https://bugzilla.mozilla.org/show_bug.cgi?id=1897907 was about firefox no longer using remotingName for the dbus name, which actually means one less side-effect.

As we can read in [the change](https://github.com/mozilla/gecko-dev/commit/a03607661537eedf1b3c8aee0b41e26507a3c280), remotingName today is basically meant to match the desktop file name, so all other side-effects should only be positive.

But let's look at all the other places that use remotingName:
* https://searchfox.org/mozilla-central/rev/0c55d51c0d2a9b672e42ad40ea54f90267f92a8e/toolkit/system/gnome/nsAlertsIconListener.cpp#198
This sets the notification hint by the remotingName set at compile time, not the environment variable. 
<sub>_Side note: It is currently wrong as it is set to `Firefox`, with capital `F`, and then mangled by libnotify4 to become `firefox_Firefox` which would almost match firefox_firefox.desktop but has wrong capitalization. But somehow both gnome-shell and kde plasma manage to trace the notification back to the right app anyways._<sub>

* https://searchfox.org/mozilla-central/rev/0c55d51c0d2a9b672e42ad40ea54f90267f92a8e/toolkit/xre/nsAppRunner.cpp#4684
The GLib `prgname` may be used in multiple places, but most importantly in Wayland it is used [to set](https://gitlab.gnome.org/GNOME/gtk/-/blob/c492e71aed030f3777ae1b5cca8ec57d12973e79/gdk/wayland/gdktoplevel-wayland.c#L876) the `xdg_toplevel::app_id` which must match the desktop file (see also: https://bugzilla.allizom.org/show_bug.cgi?id=1826330). KDE Plasma Wayland currently fails to match the Firefox window with the Firefox app in the taskbar, and instead shows another taskbar entry with a generic icon. This is currently fixed with `Exec=env BAMF_DESKTOP_FILE_HINT=/var/lib/snapd/desktop/applications/firefox_firefox.desktop` in the desktop file, but running `snap run firefox` from the terminal still presents the issue. This change also fixes that; another positive side-effect.
![Screenshot_20240809_111942](https://github.com/user-attachments/assets/e56309af-4b27-4357-978c-56b7e584d975)

* https://searchfox.org/mozilla-central/rev/0c55d51c0d2a9b672e42ad40ea54f90267f92a8e/widget/gtk/nsAppShell.cpp#386
Same thing as above but for X11 `WM_CLASS`, which is the equivalent of `xdg_toplevel::app_id`.

And that is all I could [find](https://searchfox.org/mozilla-central/search?q=remotingName)!

Reply:
PS: @stransky who maintains the Firefox RPM for Fedora also sets `MOZ_APP_REMOTINGNAME` to the desktop file name (which is `org.mozilla.firefox.desktop` there)

Reply:
> PS: @stransky who maintains the Firefox RPM for Fedora also sets `MOZ_APP_REMOTINGNAME` to the desktop file name (which is `org.mozilla.firefox.desktop` there)

yes, and he's the one who suggested it: https://bugzilla.mozilla.org/show_bug.cgi?id=1897907#c31

Reply:
@aleasto that's exactly why I'd prefer we run this by @stransky, because I'm not at ease enough and i already made mistake on that part in the past. There are also weirdness of integration wrt snapd / kde (and i dont know who is responsible) that makes this not a problem on gnome somehow?

like maybe `# The desktop file is exported by snapd as firefox_firefox.desktop` is actually what is wrong here. I think @bandali0 wanted to also check that

Reply:
> yes, and he's the one who suggested it: https://bugzilla.mozilla.org/show_bug.cgi?id=1897907#c31

Specifically for setting the bus name though. Other places still use remotingName though.

> like maybe # The desktop file is exported by snapd as firefox_firefox.desktop is actually what is wrong here. I think @bandali0 wanted to also check that

Unfortunately that's how snapd works; so far we can only work around it.

Reply:
> > yes, and he's the one who suggested it: https://bugzilla.mozilla.org/show_bug.cgi?id=1897907#c31
> 
> Specifically for setting the bus name; other places still use remotingName though.

yes, and I'm not confident in asserting this change is not going to go wreck other things in our back, hence why I'd like martin to check

> 
> > like maybe # The desktop file is exported by snapd as firefox_firefox.desktop is actually what is wrong here. I think @bandali0 wanted to also check that
> 
> Unfortunately that's how snapd works; so far we can only work around it.

i dont know where in snapd this is coming from but one thing snapd is good at is fixing and releasing things. It would be nice to verify if the actual behavior is indeed correct or a subtle bug. And fix the bug if it's the case.

Reply:
@aleasto let's be clear, it's not that I'm against the change, I'm just unsure it's the right fix. ultimately, i dont have review credentials here, but amin was not confident either so asked me :)

Reply:
> yes, and I'm not confident in asserting this change is not going to go wreck other things in our back, hence why I'd like martin to check

I agree. I've pinged them here but if you have other ways of bringing this to their attention please do.

> i dont know where in snapd this is coming from but one thing snapd is good at is fixing and releasing things. It would be nice to verify if the actual behavior is indeed correct or a subtle bug. And fix the bug if it's the case.

It is intended behaviour.